tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py

```diff
@@ -48,22 +41,35 @@ def get_configuration(request):
     return request.param
 
 
-# Tests
-
-def extra_configuration_before_yield():
-    """Make sure to delete any existing key with the same name before performing the test"""
-    for reg_key, arch in registry_list:
-        try:
-            key_h = RegOpenKeyEx(registry_parser[key], reg_key, 0, win32con.KEY_ALL_ACCESS | arch)
-            delete_registry(key_h)
-        except OSError: # Ignore the error in case the key doesn't exists
-            pass
-
-
-def extra_configuration_after_yield():
-    """Make sure to delete the key after performing the test"""
-    for key_h, _ in handle_list:
-        try:
-            delete_registry(key_h)
-        except OSError:
-            pass
+@pytest.mark.parametrize('key, subkey, arch', [
+                         (key, sub_key, win32con.KEY_WOW64_32KEY),
+                         (key, sub_key, win32con.KEY_WOW64_64KEY),
+                         (key, sub_key_2, win32con.KEY_WOW64_64KEY)
+                         ])
+def test_tags(key, subkey, arch,
+              get_configuration, configure_environment, restart_syscheckd, wait_for_fim_start):
+    """
+    Check the tags functionality by applying some tags an ensuring the events raised for the monitored directory has
+    the expected tags.
+
+    Parameters
+    ----------
+    folder : str
+        Directory where the file is being created.
+    name : str
+        Name of the file to be created.
+    content : str, bytes
+        Content to fill the new file.
+    """
+    defined_tags = get_configuration['metadata']['fim_tags']
+
+    def tag_validator(event):
+        assert defined_tags == event['data']['tags'], f'defined_tags are not equal'
+
+
+    registry_value_cud(key, subkey, wazuh_log_monitor, arch=arch,
+                     time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled',
+                     min_timeout=global_parameters.default_timeout,
+                     triggers_event=True,
+                     validators_after_cud=[tag_validator]
+                    )
```

Comment:
These lines are under-indented.
```suggestion
                       time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled',
                       min_timeout=global_parameters.default_timeout,
                       triggers_event=True,
                       validators_after_cud=[tag_validator]
                       )
```